infrastructure/sentry-subgraph/src/utils/updateChallenge.ts

```diff
@@ -38,6 +39,76 @@ export function updateChallenge(referee: Referee, challenge: Challenge): Challen
     challenge.numberOfEligibleClaimers = challengeStruct.numberOfEligibleClaimers;
     challenge.amountClaimedByClaimers = challengeStruct.amountClaimedByClaimers;
 
+    //return early here because assertion checking was not turned on until this block number
+    let firstAssertionCheckingBlockNumber = 176156007;
+    if (event.block.number.lt(BigInt.fromI32(firstAssertionCheckingBlockNumber))) {
+        return challenge;
+    }
+
+    //set challenge id range and load previous challenge entity
+    const currentChallengeId = challenge.challengeNumber;
+    const previousChallengeId = currentChallengeId.minus(BigInt.fromI32(1));
+    const previousChallenge = Challenge.load(previousChallengeId.toString());
+
+    //previous challenge not found, therefore on the first challenge
+    if (!previousChallenge) {
+        //set challenge field in NodeConfirmed event to challenge id and save
+        let nodeConfirmation = NodeConfirmation.load(challenge.assertionId.toString());
+        if (!nodeConfirmation) {
+            log.warning(`Failed to load nodeConfirmation entity with id: ${challenge.assertionId}`, []);
+            return challenge;
+        }
+        nodeConfirmation.challenge = challenge.id;
+        nodeConfirmation.save();
+        return challenge;
+    }
+
+    //execute based on gap between previous and current assertion id
+    const assertionIdGapSize = challenge.assertionId.minus(previousChallenge.assertionId);
+    if (assertionIdGapSize.equals(BigInt.fromI32(1))) { //gap size == 1 (no gap)
+        //set challenge field in this NodeConfirmation entity
+        let nodeConfirmation = NodeConfirmation.load(challenge.assertionId.toString());
+        if (!nodeConfirmation) {
+            log.warning(`Failed to load nodeConfirmation entity with id: ${challenge.assertionId}`, []);
+            return challenge;
+        }
+        nodeConfirmation.challenge = challenge.id;
+        nodeConfirmation.save();
+    } else if (assertionIdGapSize.gt(BigInt.fromI32(1))) { //gap size > 1
+        const refereeConfig = RefereeConfig.load("RefereeConfig");
+        if (!refereeConfig) {
+            log.warning("Failed to find refereeConfig", [])
+            let nodeConfirmation = NodeConfirmation.load(challenge.assertionId.toString());
+            if (!nodeConfirmation) {
+                log.warning(`Failed to load nodeConfirmation entity with id: ${challenge.assertionId}`, []);
+                return challenge;
+            }
+            nodeConfirmation.challenge = challenge.id;
+            nodeConfirmation.save();
+            return challenge;
+        }
+        if (refereeConfig.version.gt(BigInt.fromI32(6))) {
```

Comment:
Should this be less than?
If we are > Referee 6, that means we're on the new referee, correct?
But if I read this right, this code will return the challenge without accounting for a gap in assertion Ids? 
Forgive me if I am mis-interpreting.

Reply:
I think @XDapps is correct, this should be less than version 6, only after version will we allow bundled challenges.